src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,79 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void
+};
+
+export const Pagination: React.FC<Props> = ({
+  total, perPage, currentPage, onPageChange,
+}) => {
+  const currentPages = [];
+  const numberOfPages = Math.ceil(total / perPage);
+
+  for (let i = 1; i <= numberOfPages; i += 1) {
+    currentPages.push(i);
+  }
+
+  const pageClickHandler = (
+    e:React.MouseEvent<HTMLAnchorElement>,
+  ) => {
+    const el = e.target as HTMLElement;
+    const clickedPAge = Number(el.textContent);
+
+    if (Number(e.target) !== currentPage) {
+      onPageChange(clickedPAge);
+    }
+  };
```

Comment:
You can pass a page here as an argument instead of an event. This will make the function much easier
```suggestion
  const pageClickHandler = (page) => {urrentPage) {
      onPageChange(page);
  };
```

Reply:
I cant as I dont pass arguments here. React automaticlly passes the event to the function, it is a handler function